### DISCUSSION TITLE
Set tempdir's baseDir to Gradle's project.getLayout().getBuildDirectory()

### DISCUSSION THREAD

**Discussion by martinda** [spockframework/spock#1967] [Q&A] [answered]
Is there a way to pass the value of the gradle `project.getLayout().getBuildDirectory()` to Spock's `baseDir`?
The equivalent of doing this in the `src/test/resources/SpockConfig.groovy` file:
```groovy
tempdir {
    baseDir project.getLayout().getBuildDirectory()
}
```
I strongly suspect no, but maybe there is another way for those who have limited room on `/tmp`?

**Comment by Vampire** [comment, marked as answer]
If there is no other way, you can still set `project.getLayout().getBuildDirectory()` as value of some system property on the `test` task and evaluated that system property within `SpockConfig.groovy`

**Reply by martinda** [reply]
For `build.gradle`:
```groovy
test {
    systemProperty 'buildDirectory', project.getLayout().getBuildDirectory().get().getAsFile().getAbsolutePath()
}
```
For `src/test/resources/SpockConfig.groovy`:
```groovy
tempdir {
    baseDir java.nio.file.Paths.get(System.getProperty('buildDirectory'))
}
```
